src/state/session/index.tsx

```diff
@@ -363,16 +369,19 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
     async account => {
       logger.debug(`session: initSession`, {}, logger.DebugContext.session)
 
-      const agent = new BskyAgent({
-        service: account.service,
-        persistSession: createPersistSessionHandler(
+      const agent = new BskyAgent({service: account.pdsUrl || account.service})
```

Comment:
If we restore a session `{ service: A, pdsUrl: B }` into this agent, I believe it will end-up with a different session `{ service: B, pdsUrl: undefined }`.  It might be safer to assign the PDS url in the same way that it would happen via `getSession`, I believe something like this:
```ts
const agent = new BskyAgent({service: account.service})
agent.pdsUrl = account.pdsUrl ? new URL(account.pdsUrl) : undefined
agent.api.xrpc.uri = agent.pdsUrl || agent.service
```

I realize it's a little awkward 🙈 We will have a better way of thinking about this before long, it is in the works!

Reply:
Ooh, thanks for spotting! Looks like I've forgotten that moving it to the constructor will make `pdsUrl` be undefined.